File: examples/python/gen.py
"""A runnable Python example: generate synthetic microstructure and print the
first three candles.

    pip install wickra-synth
    python examples/python/gen.py

Every language example uses the same seed and prints the same candles — that is
the cross-language guarantee.
"""

import json

from wickra_synth import Synth

SPEC = json.dumps(
    {
        "seed": 42,
        "bars": 20,
        "start_price": 100.0,
        "regimes": [{"kind": "trend", "len": 20, "drift": 0.002, "vol": 0.01}],
        "microstructure": {
            "book_depth": 5,
            "spread_bps": 4.0,
            "trade_rate": 8.0,
            "funding": {"interval_bars": 8, "base_rate": 0.0001, "sensitivity": 0.5},
        },
    }
)


def main() -> None:
    synth = Synth(SPEC)
    out = json.loads(synth.command(json.dumps({"cmd": "generate"})))

    print(f"wickra-synth {Synth.version()}")
    print(f"bars: {len(out['candles'])}")
    print("first 3 candles:")
    for candle in out["candles"][:3]:
        print(f"  {json.dumps(candle, separators=(',', ':'))}")


if __name__ == "__main__":
    main()

File: examples/node/gen.js
// A runnable Node.js example: generate synthetic microstructure and print the
// first three candles.
//
//   npm install
//   node examples/node/gen.js
//
// Every language example uses the same seed and prints the same candles.
"use strict";

const { Synth } = require("wickra-synth");

const SPEC = JSON.stringify({
  seed: 42,
  bars: 20,
  start_price: 100.0,
  regimes: [{ kind: "trend", len: 20, drift: 0.002, vol: 0.01 }],
  microstructure: {
    book_depth: 5,
    spread_bps: 4.0,
    trade_rate: 8.0,
    funding: { interval_bars: 8, base_rate: 0.0001, sensitivity: 0.5 },
  },
});

const synth = new Synth(SPEC);
const out = JSON.parse(synth.command(JSON.stringify({ cmd: "generate" })));

console.log(`wickra-synth ${synth.version()}`);
console.log(`bars: ${out.candles.length}`);
console.log("first 3 candles:");
for (const candle of out.candles.slice(0, 3)) {
  console.log(`  ${JSON.stringify(candle)}`);
}
